var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -368,7 +370,9 @@ def archdir(self):
         specific directory by looking for a specific binary.
         """
         if self.spec.satisfies("+cmake"):
-            neuron_arch = str(self.spec.architecture).split("-")[2]
+            neuron_arch = subprocess.Popen(["uname", "-p"],
+                                           stdout=subprocess.PIPE) \
+                .communicate()[0].decode('UTF-8').rstrip()
```

Comment:
:vomiting_face: I guess `check_output` is not available due to Python restrictions?

Reply:
Exactly :+1:  
https://travis-ci.org/github/BlueBrain/spack/jobs/703621783
I found [here](https://stackoverflow.com/a/4760517) this way to do this

Reply:
Thinking bit more about this, I wonder if we can eliminate setting arch dir completely. Downstream packages can “find” location of special and libraries without much efforts !

Reply:
You're right that this should be possible but maybe this could be part of a refactoring of the whole `neurodamus` recipes?

Reply:
Ok..let’s not block then! Could you create ticket under BSD for neurodamus refactoring and mention this as one of the item?

Reply:
https://bbpteam.epfl.ch/project/issues/browse/BSD-154